action.yml

```
@@ -0,0 +1,44 @@
+name: 'Build Plugin'
+description: 'Build GOG Galaxy integration on current OS'
+
+inputs:
+  src:
+    description: 'Location of directory containing manifest.json and essential plugin files'
+    required: true
+    default: 'src'
+  build:
+    description: 'Relative path to expected build directory' 
+    required: true
+    default: 'build'
+  requirements:
+    description: 'Location of Python requirements file'
+    required: true
+    default: 'requirements/app.txt'
+  third-party-target:
+    description: 'Target directory where pip packages are dumped; relative to `build`'
+    required: false
+    default: '.'
+  python-arch:
+    description: 'Python binary to be used: x64 for Mac or x86 for Windows'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.7.5'  # Version 3.7.3 used by GOG not avaialable https://github.com/actions/setup-python/issues/474
+        architecture: ${{ inputs.python-arch }}
+    - run: pip install fog.buildtools
+      shell: bash
+    - name: Build plugin
+      shell: python
+      run: |
+        from fog.buildtools import build
+        build(
+          src='${{ inputs.src }}',
+          output='${{ inputs.build }}',
+          third_party_output='${{ inputs.third-party-target }}',
+          requirements='${{ inputs.requirements }}',
+        ) 
```

```suggestion
        ) 
    - name: Zip build
       uses: thedoctor0/zip-release@0.7.1
         with:
          type: 'zip'
          filename: 'build.zip'
          path: 'build'
```
might as well zip it here

---

hm, might be... firstly I wanted take out any top level decisions (like zip naming) from that action, so it has to be done on plugin side, but plugin name and uuid could be both passed on this action too... idk